### DISCUSSION TITLE
Joining a container to an Existing Active Directory

### DISCUSSION THREAD

**Discussion by empereira** [samba-in-kubernetes/samba-container#162] [General]
hi,

First of all, I would like to congratulate you on your excellent work!!!

Which of the yamls should I use to join an AD container with our main AD?

https://github.com/samba-in-kubernetes/samba-container/tree/master/examples/kubernetes

**Comment by phlogistonjohn** [comment]
Hi there, thank you for the kind words.

There's one example yaml for using AD domain controllers.
https://github.com/samba-in-kubernetes/samba-container/blob/master/examples/kubernetes/samba-ad-server-deployment.yml

But if I understand your question correctly, you have an existing AD domain and want to have the containerized DC be part of that domain. It's something I looked into very briefly and tried to make the containers capable of doing that but IIRC it's not been tested by any one in our team. I would suggest trying it out "by hand" with docker/podman and then adapting the yaml above to.  I'd be happy to assist you but I don't have a simple answer.
